masonry/src/text/render_text.rs

```diff
@@ -60,6 +61,63 @@ pub fn render_text(
                         }
                     }),
                 );
+            // We want to draw the underline/strikethrough on top of the glyphs, so we draw it later
+            if let Some(underline) = &style.underline {
+                let underline_brush = &brushes[underline.brush.0];
+                let run_metrics = glyph_run.run().metrics();
+                let offset = match underline.offset {
+                    Some(offset) => offset,
+                    None => run_metrics.underline_offset,
+                };
+                let width = match underline.size {
+                    Some(size) => size,
+                    None => run_metrics.underline_size,
+                };
+                // The `offset` is the distance from the baseline to the *top* of the underline
+                // so we move the line down by half the width
+                // Remember that we are using a y-down coordinate system
+                let y = glyph_run.baseline() - offset;
+
+                let line = Line::new(
+                    (glyph_run.offset() as f64, y as f64),
+                    ((glyph_run.offset() + glyph_run.advance()) as f64, y as f64),
+                );
+                scene.stroke(
+                    &Stroke::new(width.into()),
+                    transform,
+                    underline_brush,
+                    None,
+                    &line,
+                );
+            }
+            if let Some(strikethrough) = &style.strikethrough {
+                let strikethrough_brush = &brushes[strikethrough.brush.0];
+                let run_metrics = glyph_run.run().metrics();
+                let offset = match strikethrough.offset {
+                    Some(offset) => offset,
+                    None => run_metrics.strikethrough_offset,
+                };
+                let width = match strikethrough.size {
+                    Some(size) => size,
+                    None => run_metrics.strikethrough_size,
+                };
+                // The `offset` is the distance from the baseline to the *top* of the strikethrough
+                // so we move the line down by half the width
+                // Remember that we are using a y-down coordinate system
+                let y = glyph_run.baseline() - offset;
```

Comment:
If offset is indeed the distance between the baseline and the top of the strikethrough, you need to correct for the font's intended strikethrough width (to get its center y-coordinate), and then let the strikethrough grow in both directions with whatever actual width is chosen.

```suggestion
                let width = match strikethrough.size {
                    Some(size) => size,
                    None => run_metrics.strikethrough_size,
                };
                // The `offset` is the distance from the baseline to the *top* of the strikethrough
                // so we move the line down by half the width
                // Remember that we are using a y-down coordinate system
                let y = glyph_run.baseline() - offset + run_metrics.strikethrough_size / 2.;
```

Reply:
Yes. That is basically my plan.
I don't quite know how to incorporate this suggestion because I'm doing some other changes based on web spec reading.